### DISCUSSION TITLE
Missing parameter: username error during authentication with Keycloak server using Authorization Code Flow

### DISCUSSION THREAD

**Discussion by atiol** [keycloak/keycloak#22993] [Q&A] [answered]
Hi,
I tried to follow the docs as much as possible and also checked on other platforms to see if I missed something but I just couldn't figure out what I did wrong.
I created a keycloak client with only `standard flow` and `service accounts roles` authentication active and created one user. I am trying to implement authorization code flow to obtain access and refresh tokens however I keep getting the error: `Missing parameter: username`. Why would keycloak expect a username parameter without first showing the login page?

Scenario:
Implement Authorization Code flow by supplying the following params (Using Insomnia to make requests);

**POST** `https://my-kyecloak-server/realms/my-realm/protocol/openid-connect/auth`

**Params**
ClientId: [Client Id]
ClientSecret: [Secret]
Authorization Endpoint: https://my-keycloak-server/realms/my-realm/protocol/openid-connect/auth
RedirectURL: http://localhost*

**Result**
`{"error":"invalid_request","error_description":"Missing parameter: username"}`

Any help will be much appreciated.

**Comment by justin-tay** [comment]
The authorization endpoint should be queried using a browser which will be with a `GET` request.

So it would look something like

```
https://my-keycloak-server/realms/my-realm/protocol/openid-connect/auth?response_type=code&client_id=clientid&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback
```

You can refer to RFC 6749 for details https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.1

**Comment by atiol** [comment]
Thanks for you quick response @justin-tay however, I actually first tried it in a browser and I also tried it using next-auth library for my nextjs project but same erroneous response.

**Reply by justin-tay** [reply]
Does pasting the url I provided my answer in the browser, after replacing with your client id not work?

**Reply by atiol** [reply, marked as answer]
It didn't work unfortunately. However, I created a new client and replaced the credentials and it worked. Could it be that keycloak failed to clear it's cache coz I started with Direct Grant then later switched to standard flow. But it seems keycloak server still used the old configuration settings and hence the need for 'username' parameter. Just a thought, I could be wrong but testing with a new client solves the issue as also pointed out by some other people in other forums.
I'll close the discussion now, thanks again.
